spec/models/i14y_search_spec.rb

```diff
@@ -215,14 +215,24 @@
 
   context 'when there is some problem with the i14y client' do
     before do
-      allow(I14yCollections).to receive(:search).and_raise Faraday::ClientError.new(Exception.new('problem'))
+      stub_request(:get, %r{api/v1/collections}).
+        to_raise(Faraday::ClientError.new('problem'))
       allow(Rails.logger).to receive(:error)
     end
 
     it 'logs the error' do
       i14y_search.run
       expect(Rails.logger).to have_received(:error).with(/I14y search problem/)
     end
+
+    # semi-integration spec to confirm that we send i14y search client errors to Datadog
+    it 'sends the information to datadog' do
+      i14y_search.run
+      expect(a_request(
+        :post,
+        'https://api.datadoghq.com/api/v1/events?api_key=datadogapikey'
```

Comment:
Wouldn't this test fail locally if the dev has an actual API key set?

Reply:
Yes -  that's an example of why devs should copy `config/secrets.yml.dev` into `secrets.yml` before running the specs: https://github.com/GSA/search-gov#service-credentials-how-we-protect-secrets

Devs should only be using production API keys if they need to interact with a production service or if they need to record external interactions in new VCR cassettes. Otherwise, they should be using the dummy values from `config/secrets.yml.dev` (which is what will be used in CircleCI).